sdk/python/packages/flet-core/src/flet_core/local_authentication/local_authentication.py

```diff
@@ -0,0 +1,137 @@
+from .mac_authentication import MacLocalAuth
+from .linux_authentication import LinuxLocalAuth
+
+
+class LocalAuthentication(Control):
+    """
+    A control to use native authentication. Works on iOS, Android, Windows, MacOS, Linux. Based on local_auth Flutter widget (https://pub.dev/packages/local_auth).
+    LocalAuthentication control is non-visual and should be added to `page.overlay` list.
+    Example:
+    ```
+    import flet as ft
+    def main(page: ft.Page):
+        auth = ft.LocalAuthentication()
+        def on_click(e):
+            print(
+                auth.authenticate(
+                    title="title", biometricsOnly=False
+                )
+            )
+        page.overlay.append(auth)
+        page.add(ft.SafeArea(content=ft.TextButton(text="authenticate", on_click=on_click)))
+    ft.app(main)
+    ```
+    """
+
+    def __init__(
+        self,
+        ref: Optional[Ref] = None,
+        data: Any = None,
+    ):
+        Control.__init__(
+            self,
+            ref=ref,
+            data=data,
+        )
+
+    def _get_control_name(self):
+        return "localauthentication"
+
+    def before_update(self):
+        self.platform = self.page.platform
+        # Both macOS and Linux have platform dependant dependencies.
+        # Hence initiating MacLocalAuth and LinuxLocalAuth after checking the platform
+        if self.platform == PagePlatform.MACOS:
+            self.maclocalauth = MacLocalAuth()
+        elif self.platform == PagePlatform.LINUX:
+            self.linuxlocalauth = LinuxLocalAuth()
+        return super().before_update()
+
+    def available(self, wait_timeout: Optional[int] = 5) -> bool:
+        if self.platform == PagePlatform.MACOS:
+            sr = self.maclocalauth.is_available()
+        elif self.platform == PagePlatform.LINUX:
+            # linux always has a user with password.
+            # Even if not so, a popup asking to authenticate without asking password will be shown
+            sr = True
+        else:
+            sr = (
+                self.invoke_method(
+                    "available", wait_for_result=True, wait_timeout=wait_timeout
+                )
+                == "true"
+            )
+
+        return sr
+
+    async def available_async(self, wait_timeout: Optional[int] = 5) -> bool:
+        if self.platform == PagePlatform.MACOS:
+            sr = await self.maclocalauth.is_available()
+        elif self.platform == PagePlatform.LINUX:
+            # linux always has a user with password.
+            # Even if not so, a popup asking to authenticate without asking password will be shown
+            sr = True
+        else:
+            sr = (
+                await self.invoke_method_async(
+                    "available", wait_for_result=True, wait_timeout=wait_timeout
+                )
+                == "true"
+            )
+
+        return sr
+
+    def authenticate(
+        self,
+        title: str,
```

Comment:
The `title` should be optional or alternatively have a default, because the Linux seems not to require it.

Reply:
I also think we should rename this argument to `reason`, as it clearer.

Reply:
will set the default to `"Please Authenticate to continue"` and rename the argument to `reason`